### DISCUSSION TITLE
New Dashboard: Supercharging session chart

### DISCUSSION THREAD

**Discussion by gryghst72** [teslamate-org/teslamate#205] [Ideas]
(Problem) It has come to many peoples attention that Tesla has changed the Supercharging curve for later models, and may end up doing that for current and future models as Tesla updates the firmware over time.

(Solution) It would be nice to see a single chart that shows rate of charge vs % state of charge for all supercharge sessions superimposed over each other with filters by software version (using colors).  Maybe even have a second graph superimposing time spent charging vs % of charge to quickly see if it's taking longer to charge, again filtering by s/w version.

If it's ok to reference another Tesla logging platform, ABRP has exactly this sort of graph, but on a fleet scaling [here](https://forum.abetterrouteplanner.com/blogs/entry/30-tesla-supercharging-summer-2019-update/).  Users of Teslamate could see something has changed for the worse and open a dialog with Tesla about it.

(Alternatives) Any/all suggested alternatives are welcome.

-Regards

**Comment by adriankumpf** [comment]
Great idea! The necessary data is already stored by TeslaMate[1]. Someone would just have to create a dashboard with those charts. PRs are welcome :)

[1]: To get all supercharging sessions the following columns of the tables "charges" can be used: 
* `fast_charger_present` (TRUE) 
* `fast_charger_brand` (Tesla)

**Comment by timguy99** [comment]
I'm not familiar with Grafana but want to jump in one day to see what sort of graphs I can create. How do I know what sort of data is available to graph? I suppose I need to look at the Teslamate Postgres DB at the very least.

But any pointers or reference that I should look into that can be suggested here?

**Comment by baylanger** [comment]
@timguy99 I'm in the same situation as you ... I want to make some changes.

grafana.com should have everything you need to learn.  Here's a screenshot of default available visualization.

![2020-02-12_23-35-11](https://user-images.githubusercontent.com/5240348/74401788-557d8280-4df0-11ea-9a0e-968163fdcae4.png)

If you want to know what data is available in the DB, there's likely a psql command that can describe all tables within the database.
